handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -244,8 +244,12 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                                             final String hostname = in.toString(offset, serverNameLength,
                                                                                 CharsetUtil.UTF_8);
 
-                                            select(ctx, IDN.toASCII(hostname,
-                                                                    IDN.ALLOW_UNASSIGNED).toLowerCase(Locale.US));
+                                            try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)